collector/perfos.go

```diff
@@ -0,0 +1,132 @@
+// returns data points from Win32_PerfRawData_PerfOS_System class (undocumented)
+
+package collector
+
+import (
+	"log"
+
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	Factories["perfos"] = NewPerfOSCollector
+}
+
+// A PerfOSCollector is a Prometheus collector for WMI metrics
+type PerfOSCollector struct {
+	ContextSwitchesPersec     *prometheus.Desc
+	ExceptionDispatchesPersec *prometheus.Desc
+	ProcessorQueueLength      *prometheus.Desc
+	SystemCallsPersec         *prometheus.Desc
+	SystemUpTime              *prometheus.Desc
+	Threads                   *prometheus.Desc
+}
+
+// NewPerfOSCollector ...
+func NewPerfOSCollector() (Collector, error) {
+	const subsystem = "perfos"
```

Comment:
Sorry about my constant naming nitpicking... But shouldn't this be `system` or possibly `os_system` (which is a bit wierd, but still)?
The "base name" of the raw metrics is `Win32_PerfRawData_Perf`, so the specific class here is really `OS_System`